src/gui/owncloudgui.cpp

```diff
@@ -168,11 +192,11 @@ void ownCloudGui::slotTrayClicked(QSystemTrayIcon::ActivationReason reason)
     if (reason == QSystemTrayIcon::Trigger) {
         if (OwncloudSetupWizard::bringWizardToFrontIfVisible()) {
             // brought wizard to front
-        } else if (_shareDialogs.size() > 0) {
-            // Share dialog(s) be hidden by other apps, bring them back
-            Q_FOREACH (const QPointer<ShareDialog> &shareDialog, _shareDialogs) {
-                Q_ASSERT(shareDialog.data());
-                raiseDialog(shareDialog);
+        } else if (!_visibleDialogs.isEmpty()) {
+            // Dialog(s) be hidden by other apps, bring them back (e.g. Share / Settings / Auth dialogs)
+            Q_FOREACH (const QPointer<QDialog> &dialog, _visibleDialogs) {
```

Comment:
Would be a good opportunity to switch that Q_FOREACH to a proper range for (don't forget to use qAsConst on _visibleDialogs though).